### ISSUE TITLE
RuntimeError: Cannot infer type from null for SQLite

### ISSUE THREAD

**Issue by eloyfelix** [sfu-db/connector-x#211]
version: 0.2.4a2

small db example attached to reproduce the issue. [sample.db.zip](https://github.com/sfu-db/connector-x/files/7789991/sample.db.zip)

It seems that is not able to process nulls from `NUMERIC` type:

`select standard_value from activities limit 1`

It might be another diferent bug but it also throws a different error when querying the whole table (85 rows).

`select standard_type, standard_value from activities`

`RuntimeError: Invalid column type Real at index: 1, name: standard_value`

the table is extracted from the ChEMBL SQLite dump https://ftp.ebi.ac.uk/pub/databases/chembl/ChEMBLdb/latest/

**Comment by wangxiaoying** [comment]
Hi @eloyfelix , thanks for sample datasaet.

We do have some type infer issue for SQLite. It is because SQLite adopts [dynamic type system](https://www.sqlite.org/datatype3.html), which means that there are two types: a declarative type (the one specified for a column when creating the table) and a data type (the true data value type), and these two types might not be able to matched. When we get the result value, its declarative type might be empty if the value is derived from some computation instead of directly select from table, and the data type might also be unavailable for NULL values.

Here is how we infer the type in connectorx currently: We issue a limit 1 query to get the first row of the result and infer the type from this row:
1. If the declarative type is specified, we will infer from it directly using the rules defined in [here](https://www.sqlite.org/datatype3.html#affname).
2. If the declarative type is not specified or cannot be matched, we will use the data type
3. If the declarative type is not specified or cannot be matched, and the data type is NULL, we will raise an error.

For you queries, since they are directly selecting columns, the declarative type should works in step 1. I checked your sample.db, and it turns out your `standard_value` is defined as `NUMERIC`, which is not currently [supported](https://github.com/sfu-db/connector-x/blob/main/Types.md#sqlite) in connectorx yet since it may contains a [different data types](https://www.sqlite.org/datatype3.html#affname) for a single column, which we don't support for the dataframes. I also checked the values of `standard_value` in the sample and it indeed contains both `INTEGER` and `REAL` types. I think one workaround is to convert this column to `REAL` type and then query it (also change the declaration in order to avoid the NULL value issue).
